packages/hardhat-core/src/internal/util/bigint.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  BigIntLike,
+  bufferToBigInt,
+  bufferToHex,
+  toBuffer,
+} from "@ethereumjs/util";
+
+function min(x: bigint, y: bigint): bigint {
+  return x < y ? x : y;
+}
+
+function max(x: bigint, y: bigint): bigint {
+  return x > y ? x : y;
+}
+
+function isBigInt(x: unknown): x is bigint {
+  return typeof x === "bigint";
+}
+
+function divUp(x: bigint, y: bigint): bigint {
+  let result = x / y;
+
+  if (x % y !== 0n) {
+    result = result + 1n;
+  }
+
+  return result;
+}
+
+function cmp(a: bigint, b: bigint): number {
+  return a < b ? -1 : a > b ? 1 : 0;
+}
+
+function mapNumberToBigInt<T>(x: number | T): bigint | T {
```

Comment:
This is a surprising function. Same for the following ones. Maybe the name is surprising?

Reply:
It's mainly about mapping types like `number | undefined` to `bigint | undefined`. The problem is that some of those types have more types (e.g. `number | Buffer | undefined`), so I wrote a generic thing.

Agree that the name is not great though. Any suggestions?

Reply:
I ended up removing this function. It was barely used after the config types changes.